tests/test_process_elife_zip_dag.py

```diff
@@ -45,60 +48,153 @@ def test_get_expected_elife_article_name_raises_exception_if_zip_name_is_malform
         assert str(error.value) == msg
 
 
+def test_get_article_from_previous_task(context):
+    # setup
+    test_asset_path = str(get_asset('elife-00666.xml').absolute())
+    article_xml = etree.tostring(etree.parse(test_asset_path))
+    add_return_value_from_previous_task(return_value=article_xml, context=context)
+    # test
+    returned_value = get_article_from_previous_task(context)
+    assert etree.tostring(returned_value) == article_xml
+
+
+@pytest.mark.parametrize('return_value', [b'', '', None])
+def test_get_article_from_previous_task_raises_exception(return_value, context):
+    # setup
+    add_return_value_from_previous_task(return_value, context)
+    message = 'Article bytes were not passed from task previous_task'
+    # setup
+    with pytest.raises(AssertionError) as error:
+        get_article_from_previous_task(context)
+        assert str(error.value) == message
```

Comment:
here it's unclear whether `AssertionError` is raised by `get_article_from_previous_task` (which is what is being tested) or from the following `assert`?

Reply:
Maybe it stems from the fact that the `assert` should be outside the `with`, otherwise it won't be reached?

Reply:
fixed.